lib/FusionInventory/Agent/Tools/Hardware.pm

```diff
@@ -531,6 +583,18 @@ sub getDeviceFullInfo {
         delete $device->{PORTS};
     }
 
+    # convert components hashref to an arrayref, sorted by index
+    my $components = $device->{COMPONENTS}->{COMPONENT};
+    if ($components && %$components) {
+        $device->{COMPONENTS}->{COMPONENT} = [
+            map { $components->{$_} }
+            sort { $a <=> $b }
+            keys %{$components}
+        ];
+    } else {
+        delete $device->{COMPONENTS};
+    }
```

Comment:
unless they are real case where this key could be populated with invalid content, I'd rather use a different existence strategy, so as to avoid auto-vivication and the need to delete it thereafter, ie:
```
if (exists $device->{COMPONENTS}) {
    $device->{COMPONENTS} = ...
}
# no need to delete anything
```

Reply:
I use the same code behind for _PORTS_

Reply:
Above :D